tests/garage/tf/regressors/test_bernoulli_mlp_regressor_with_model.py

```diff
@@ -0,0 +1,233 @@
+import pickle
+from unittest import mock
+
+from nose2.tools.params import params
+import numpy as np
+import tensorflow as tf
+
+from garage.tf.optimizers import ConjugateGradientOptimizer, LbfgsOptimizer
+from garage.tf.regressors import BernoulliMLPRegressorWithModel
+from tests.fixtures import TfGraphTestCase
+
+
+def get_labels(input_shape, xs):
+    label = [0, 0]
+    if input_shape == (1, ):
+        ys = 0 if np.sin(xs) <= 0 else 1
+        label[ys] = 1
+
+    elif input_shape == (2, ):
+        ys = int(np.round(xs[0])) ^ int(np.round(xs[1]))
+        label[ys] = 1
+
+    return label
+
+
+def get_train_data(input_shape):
+    if input_shape == (1, ):
+        data = np.linspace(-np.pi, np.pi, 1000)
+        obs = [{
+            'observations': [[x]],
+            'returns': [get_labels(input_shape, x)]
+        } for x in data]
+
+    elif input_shape == (2, ):
+        data = [np.random.rand(2) for _ in range(1000)]
+        obs = [{
+            'observations': [x],
+            'returns': [get_labels(input_shape, x)]
+        } for x in data]
+    return obs
+
+
+def get_test_data(input_shape):
+    if input_shape == (1, ):
+        paths = {
+            'observations': [[-np.pi / 2], [-np.pi / 3], [-np.pi / 4],
+                             [np.pi / 4], [np.pi / 3], [np.pi / 4]]
+        }
+        expected = [[1, 0], [1, 0], [1, 0], [0, 1], [0, 1], [0, 1]]
+
+    elif input_shape == (2, ):
+        paths = {'observations': [[0, 0], [0, 1], [1, 0], [1, 1]]}
+        expected = [[1, 0], [0, 1], [0, 1], [1, 0]]
+
+    return paths, expected
+
+
+class TestBernoulliMLPRegressorWithModel(TfGraphTestCase):
+    @params(((1, ), 2), ((2, ), 2))
+    def test_fit_normalized(self, input_shape, output_dim):
+        bmr = BernoulliMLPRegressorWithModel(
+            input_shape=input_shape, output_dim=output_dim)
+        obs = get_train_data(input_shape)
+
+        observations = np.concatenate([p['observations'] for p in obs])
+        returns = np.concatenate([p['returns'] for p in obs])
+        returns = returns.reshape((-1, 2))
+
+        for _ in range(150):
+            bmr.fit(observations, returns)
+
+        paths, expected = get_test_data(input_shape)
+
+        prediction = bmr.predict(paths['observations'])
+
+        assert np.allclose(prediction, expected, rtol=0, atol=0.1)
+
+        x_mean = self.sess.run(bmr.model.networks['default'].x_mean)
+        x_mean_expected = np.mean(observations, axis=0, keepdims=True)
+        x_std = self.sess.run(bmr.model.networks['default'].x_std)
+        x_std_expected = np.std(observations, axis=0, keepdims=True)
+
+        assert np.allclose(x_mean, x_mean_expected)
+        assert np.allclose(x_std, x_std_expected)
+
+    @params(((1, ), 2), ((2, ), 2))
```

Comment:
what if output_dim is smaller than flat(input_shape)?

Reply:
will update with a test